src/tiled/automapper.cpp

```diff
@@ -809,11 +809,18 @@ void AutoMapper::copyTileRegion(const TileLayer *srcLayer, int srcX, int srcY,
                                 int width, int height,
                                 TileLayer *dstLayer, int dstX, int dstY)
 {
-    const int startX = qMax(dstX, 0);
-    const int startY = qMax(dstY, 0);
+    int startX = dstX;
+    int startY = dstY;
 
-    const int endX = qMin(dstX + width, dstLayer->width());
-    const int endY = qMin(dstY + height, dstLayer->height());
+    int endX = dstX + width;
+    int endY = dstY + height;
+
+    if (!mMapWork->infinite()) {
+        startX = qMax(dstX, startX);
```

Comment:
`dstX` and `startX` are the same at this point. You meant `0` instead of `dstX`.